tools/make.py

```diff
@@ -54,291 +49,343 @@
 
-    # Test
-    build_data_blob = {} if options.build_data else None
-    for test_no in p:
-        test = Test(test_no)
-        if options.automated is not None:    test.automated = options.automated
-        if options.dependencies is not None: test.dependencies = options.dependencies
-        if options.host_test is not None:    test.host_test = options.host_test;
-        if options.peripherals is not None:  test.peripherals = options.peripherals;
-        if options.duration is not None:     test.duration = options.duration;
-        if options.extra is not None:        test.extra_files = options.extra
 
-        if not test.is_supported(mcu, toolchain):
-            print('The selected test is not supported on target %s with toolchain %s' % (mcu, toolchain))
-            sys.exit()
+        notify = TerminalNotifier(options.verbose, options.silent, options.color)
 
-        # Linking with extra libraries
-        if options.rpc:      test.dependencies.append(RPC_LIBRARY)
-        if options.usb:      test.dependencies.append(USB_LIBRARIES)
-        if options.dsp:      test.dependencies.append(DSP_LIBRARIES)
-        if options.testlib:  test.dependencies.append(TEST_MBED_LIB)
+        if not TOOLCHAIN_CLASSES[toolchain].check_executable():
+            search_path = TOOLCHAIN_PATHS[toolchain] or "No path set"
+            args_error(parser, "Could not find executable for %s.\n"
+                               "Currently set search path: %s"
+                               %(toolchain, search_path))
 
-        build_dir = join(BUILD_DIR, "test", mcu, toolchain, test.id)
         if options.source_dir is not None:
-            test.source_dir = options.source_dir
-            build_dir = options.source_dir
-
-        if options.build_dir is not None:
-            build_dir = options.build_dir
-
-        try:
-            bin_file, update_file = build_project(
-                test.source_dir,
-                build_dir,
+            wrapped_build_project(
+                options.source_dir,
+                options.build_dir,
                 mcu,
                 toolchain,
-                set(test.dependencies),
-                linker_script=options.linker_script,
-                clean=options.clean,
                 notify=notify,
-                report=build_data_blob,
-                macros=options.macros,
-                jobs=options.jobs,
-                name=options.artifact_name,
-                app_config=options.app_config,
-                inc_dirs=[dirname(MBED_LIBRARIES)],
                 build_profile=extract_profile(parser, options, toolchain),
-                stats_depth=options.stats_depth,
-                ignore=options.ignore
+                **default_args_dict(options)
             )
-            if update_file:
-                print('Update Image: %s' % update_file)
-            print('Image: %s' % bin_file)
-
-            if options.disk:
```

Comment:
Looks like the flashing, reset, and a serial output was removed as well. I'm totally fine with this, especially since this is handled by Mbed CLI now. Just making sure the removal was for similar reasons.

Reply:
That's why they were removed.